src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AsyncLazy`1.cs

```diff
@@ -215,20 +248,69 @@ public T GetValue(CancellationToken cancellationToken)
             // but we don't want multiple threads attempting to compute the same thing.
             if (request != null)
             {
+                // Always want to make sure that if we ourselves are cancelling this request, that the request is cleaned up.
                 request.RegisterForCancellation(OnAsynchronousRequestCancelled, cancellationToken);
```

Comment:
i believe this must still exist in some form.  note: we could consider specializing this to be sync or async in the blocks below.  but it was breaking my brain, so i kept it here.